libs/server/Auth/Settings/AuthenticationSettings.cs

```diff
@@ -34,7 +34,12 @@ public enum GarnetAuthenticationMode
         /// <summary>
         /// ACL mode using Aad token instead of password. Here username is expected to be ObjectId or a valid Group's Object Id and token will be validated for claims.
         /// </summary>
-        AclWithAad
+        AclWithAad,
+
+        /// <summary>
+        /// Custom authentication mode. Requires a custom authenticator to be implemented and passed during Garnet server instantiation.
+        /// </summary>
+        Custom
```

Comment:
I don't think we should expose the custom option, because the user cannot provide an IAuthenticationSettings command line parameter. It will be confusing and may create bugs.